JetStreamDriver.js

```diff
@@ -1198,58 +1205,42 @@ class WasmEMCCBenchmark extends AsyncBenchmark {
                 },
             };
             globalObject.Module = Module;
-            `;
-        return str;
+        `;
     }
 
-    // FIXME: Why is this part of the runnerCode and not prerunCode?
+    // This is in runnerCode rather than prerunCode because prerunCode isn't currently structured to be async by default.
     get runnerCode() {
-        let str = `function loadBlob(key, path, andThen) {`;
-
+        let str = `(async function doRunWrapper() {`
         if (isInBrowser) {
             str += `
-                var xhr = new XMLHttpRequest();
-                xhr.open('GET', path, true);
-                xhr.responseType = 'arraybuffer';
-                xhr.onload = function() {
-                    Module[key] = new Int8Array(xhr.response);
-                    andThen();
-                };
-                xhr.send(null);
+                async function getBinary(key, blobURL) {
+                    const response = await fetch(blobURL);
```

Comment:
As I said above, this `getBinary` function is only used for the Wasm binary of Emscripten benchmarks, but could we use some more general mechanism from the JavaScript line items?